src/lib.rs

```diff
@@ -303,11 +310,24 @@ impl RegisterArrayInfo {
     }
 }
 
+impl RegisterClusterInfo {
+    fn parse(tree: &Element) -> RegisterClusterInfo {
+        RegisterClusterInfo {
+            registers: Some(tree.children.iter()
+                        .filter(|c| c.name == "register")
+                        .filter_map(Register::parse)
+                        .collect())
+        }
+    }
+}
+
 impl Register {
     // TODO handle "clusters", return `Register` not an `Option`
     fn parse(tree: &Element) -> Option<Register> {
```

Comment:
Then this could change to parse the *contents* of a `<register>` tree and there would be a *function* that parses either a `<register>` or a `<cluster>` that calls `Register::parse` or `Cluster::parse` depending on the tag.